### PR TITLE
fix: add tests and paramStructure support

### LINKED ISSUE #14: support for by-position and by-params
`eth_getBlock, [0, 1]`
`eth_getBlock {blockNumber: 1, from: 0}`

### PR DESCRIPTION
**Pull request by BelfordZ**
fixes #14 #12 #15

### CODE REVIEW

File: src/generate-method-mapping.js [hunk 1]
```diff
@@ -12,12 +13,25 @@ const generateResponse = async (schema, methodName) => {
...
-    result[methodObject.name] = (args, cb) => {
-      return cb(null, generateResponse(schema, methodObject.name));
-    };
-    return result;
+const makeHandler = (paramStructure, openrpcSchema, name, validator) => {
+  return (args, cb) => {
+    if (paramStructure === 'by-name') {
+    } else {
+      console.log(args);
+    }
```
Comment (shanejonas) [inline src/generate-method-mapping.js]: ?

Comment (BelfordZ) [reply src/generate-method-mapping.js]: thats how you handle it right? stdout pipe server ftw!

File: package.json [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
+  },
   "dependencies": {
     "@open-rpc/schema-utils-js": "^1.0.1",
+    "ajv": "^6.10.0",
     "jayson": "^2.1.2",
     "json-schema-faker": "^0.5.0-rc16"
   },
-  "devDependencies": {},
+  "devDependencies": {
+    "@open-rpc/examples": "^1.0.0",
```
Comment (shanejonas) [inline package.json]: bump dat

File: src/generate-method-mapping.js [hunk 3]
```diff
@@ -1,23 +1,50 @@
 const jsf = require('json-schema-faker');
+const Ajv = require('ajv');
+const _ = require('lodash');
 
-const generateResponse = async (schema, methodName) => {
-  const method = schema.methods[methodName];
-
-  if (method === undefined) { return 'method not found: ' + methodName; }
+const generateResponse = async (method) => {
+  if (method === undefined) { return 'method not found: ' + method.name; }
```
Comment (shanejonas) [inline src/generate-method-mapping.js]: should this throw or something instead of return a string?

Comment (BelfordZ) [reply src/generate-method-mapping.js]: #18

**Pull request merged**